### PR TITLE
CC-69 Build a static version of News page in React

### PR DESCRIPTION
**Pull request by PavelPyasecky**
[CC-69](https://pavelpiasetski.atlassian.net/browse/CC-69)

### CODE REVIEW

File: src/components/Content/Content.tsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
 import React from 'react'
-import { Home } from './Home/Home'
 import styled from 'styled-components'
+import {Outlet} from "react-router-dom";
 
 type ContentProps = {}
 
 const StyledContent = styled.div`
-    padding-left: 85px;
+    padding-left: 95px;
```
Comment (17Sergey) [inline src/components/Content/Content.tsx]: Please adapt to new main version, where padding is set with variables

File: src/components/Sidebar/MenuItems/MenuItem/MenuItemHeader/MenuItemHeader.tsx
Comment (17Sergey) [inline src/components/Sidebar/MenuItems/MenuItem/MenuItemHeader/MenuItemHeader.tsx:1]: Please adapt it for current version of Sidebar(without tailwind and with everything new)

File: src/components/service/RouteError.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+export const RouteError: React.FC<RouteErrorProps> = ({}) => {
+    const error:RouteErrorObject = useRouteError() as RouteErrorObject;
+    console.error(error);
+
+    return (
+        <div>
+            <h1>Oops!</h1>
+            <p>Sorry, an unexpected error has occurred. </p>
```
Comment (17Sergey) [inline src/components/service/RouteError.tsx]: I think it's better to add Russian words

File: src/main.tsx [hunk 3]
```diff
@@ -7,13 +7,45 @@ import App from "./App"
...
 import { ErrorBoundary } from "react-error-boundary"
-import { AppErrorFallback } from "./components/common/AppErrorFallback/AppErrorFallback"
+import { AppErrorFallback } from "./components/service/AppErrorFallback/AppErrorFallback"
+import type {Router as RemixRouter} from "@remix-run/router/dist/router";
+import {createBrowserRouter, RouterProvider} from "react-router-dom";
+import {Home} from "./components/Content/Home/Home";
+import {RouteError} from "./components/service/RouteError";
+import {News} from "./components/Content/News/News";
+
+const router: RemixRouter = createBrowserRouter([
```
Comment (17Sergey) [inline src/main.tsx]: Great! I've heard about Remix, but didn't know it worked with React! Nice job!

File: src/components/Sidebar/MenuItems/MenuItem/MenuItemHeader/MenuItemHeader.tsx
Comment (17Sergey) [reply src/components/Sidebar/MenuItems/MenuItem/MenuItemHeader/MenuItemHeader.tsx:1]: (Current will be when my pr is accepted ^))

File: src/components/Content/News/NewsDetails/NewsDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    grid-area: text;
+`;
+
+const NewsCreatedDate = styled.div`
+    text-align: right;
+    
+    grid-area: date;
+`;
+
+const NewsAssetImage = styled(Image)`
```
Comment (17Sergey) [inline src/components/Content/News/NewsDetails/NewsDetails.tsx]: Don't you think it's better to start name of styled-components with 'Styled'. Just to know what it is and what it does. I think I've seen this in doc

File: src/components/Content/News/NewsDetails/NewsDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+const NewsItemDetailsWrapper = styled.div`
+    flex: 25rem;
+
+    overflow-y: scroll;
+    height: 100vh;
+
+    display: grid;
+    grid-template-columns: 48% 48%;
```
Comment (17Sergey) [inline src/components/Content/News/NewsDetails/NewsDetails.tsx]: Ohhhhh I think I know how to easily animate width of news details so it's like it's flying from the right :) Just like in Sidebar new version. Remind me later if you have problems with implementation.

File: src/components/Content/News/NewsDetails/NewsDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+type NewsItemDetailsProps = {}
+
+export const NewsItemDetails: React.FC<NewsItemDetailsProps> = ({}) => {
+    const parseDate = (date: string): string => date.slice(0, 10).split("-").reverse().join(".");
+
+    return (
+        <NewsItemDetailsWrapper>
+            <NewsMainImageWrapper>
+                <Image
```
Comment (17Sergey) [inline src/components/Content/News/NewsDetails/NewsDetails.tsx]: Why did you use Image component of 'antd'? Maybe any benefits that we might take for our whole app?

**Review by 17Sergey — CHANGES_REQUESTED** [review]

File: src/components/service/RouteError.tsx (hunk 2 above)
Comment (17Sergey) [reply src/components/service/RouteError.tsx]: By the way, one of best UX design practices is to add some actions to the Error page. Like a link to main page where there is no error

**Comment by ArtsemiLasyi** [comment]
Hey, @PavelPyasecky, I don't think we need to have left side of the page (list of news). It can be good way to manage news in _admin panel_ for example, but not for common visitors b/c it makes page overly loaded by unnecessary content.
I think we need to find some other ways to show news (introduce working with search bar or calendar for example).
What do you think?
image
cc @17Sergey

**Comment by PavelPyasecky** [comment]
> Hey, @PavelPyasecky, I don't think we need to have left side of the page (list of news). It can be good way to manage news in _admin panel_ for example, but not for common visitors b/c it makes page overly loaded by unnecessary content. I think we need to find some other ways to show news (introduce working with search bar or calendar for example). What do you think? image cc @17Sergey

@ArtsemiLasyi In my view, This is a modern approach allowing us to save one extra click during news surfing. The idea adding search bar is really good and it's going to be added further! Thanks!

File: src/components/Content/News/NewsDetails/NewsDetails.tsx (hunk 6 above)
Comment (PavelPyasecky) [reply src/components/Content/News/NewsDetails/NewsDetails.tsx]: There are many preview options. 
image

File: src/components/Content/News/NewsDetails/NewsDetails.tsx (hunk 5 above)
Comment (PavelPyasecky) [reply src/components/Content/News/NewsDetails/NewsDetails.tsx]: I can't say that I understand the key point of the comment. Could you please say it in other words?)

**Comment by ArtsemiLasyi** [comment]
@PavelPyasecky, let's discuss everything in call

**Comment by ArtsemiLasyi** [comment]
@PavelPyasecky, let's discuss everything in call

File: src/components/Content/News/NewsDetails/NewsDetails.tsx (hunk 5 above)
Comment (17Sergey) [reply src/components/Content/News/NewsDetails/NewsDetails.tsx]: If we decide to animate the News details coming from the right side, when we click a news in News list. I might know how to animate that

File: src/components/Content/News/NewsDetails/NewsDetails.tsx (hunk 6 above)
Comment (17Sergey) [reply src/components/Content/News/NewsDetails/NewsDetails.tsx]: Got it. Great solution for our case.

File: src/components/Content/News/NewsDetails/NewsDetails.tsx (hunk 4 above)
Comment (PavelPyasecky) [reply src/components/Content/News/NewsDetails/NewsDetails.tsx]: I have totally forgotten about that. Thank you for reminding!

File: src/components/Content/News/NewsDetails/NewsDetails.tsx (hunk 5 above)
Comment (PavelPyasecky) [reply src/components/Content/News/NewsDetails/NewsDetails.tsx]: Ooh, yeah, nice idea! I will try to implement that)

**Review by 17Sergey — APPROVED** [review]
Great job!

**Comment by ArtsemiLasyi** [comment]
@PavelPyasecky, your new approach for news page is much better! Great job! :)
I approve it. Please, resolve merge conflicts and feel free to merge

**Review by ArtsemiLasyi — APPROVED** [review]

**Pull request merged**